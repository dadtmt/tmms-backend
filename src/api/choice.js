import R from 'ramda'

export const isInteractive = R.cond([
  [
    R.propEq('type', 'dice'),
    R.pipe(R.path(['content', 'master']), R.not)
  ],
  [R.propEq('type', 'characterSheet'), R.F],
  [R.T, R.T]
])

export const newChoice = values => ({
  ...values,
  interactive: isInteractive(values),
  made: false
})
