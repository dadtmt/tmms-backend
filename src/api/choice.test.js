import { isInteractive, newChoice } from './choice'

describe('isInteractive', () => {
  it('return true by default', () => {
    const values = {}
    expect(isInteractive(values)).toBeTruthy()
  })
  it('return false if type is dice and master true', () => {
    const values = {
      content: {
        master: true
      },
      type: 'dice'
    }
    expect(isInteractive(values)).toBeFalsy()
  })
  it('return true if type is dice and master false', () => {
    const values = {
      content: {
        master: false
      },
      type: 'dice'
    }
    expect(isInteractive(values)).toBeTruthy()
  })
  it('return false if type is characterSheet', () => {
    const values = {
      type: 'characterSheet'
    }
    expect(isInteractive(values)).toBeFalsy()
  })
})

describe('newChoice', () => {
  it('copy values and add interactive and made choice', () => {
    const values = {
      some: 'values'
    }
    expect(newChoice(values)).toMatchSnapshot()
  })
  it('copy values and add interactive false for characterSheet type', () => {
    const values = {
      some: 'values',
      type: 'characterSheet'
    }
    expect(newChoice(values)).toMatchSnapshot()
  })
  it(
  'copy values and add interactive false for dice type and content.master true',
    () => {
      const values = {
        content: {
          master: true
        },
        some: 'values',
        type: 'dice'
      }
      expect(newChoice(values)).toMatchSnapshot()
    }
  )
})
