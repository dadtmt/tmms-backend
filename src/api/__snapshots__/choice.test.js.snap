exports[`newChoice copy values and add interactive and made choice 1`] = `
Object {
  "interactive": true,
  "made": false,
  "some": "values",
}
`;

exports[`newChoice copy values and add interactive false for characterSheet type 1`] = `
Object {
  "interactive": false,
  "made": false,
  "some": "values",
  "type": "characterSheet",
}
`;

exports[`newChoice copy values and add interactive false for dice type and content.master true 1`] = `
Object {
  "content": Object {
    "master": true,
  },
  "interactive": false,
  "made": false,
  "some": "values",
  "type": "dice",
}
`;
